internal/util/xrh_header_parser.go

```diff
@@ -0,0 +1,81 @@
+package util
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+	"github.com/sirupsen/logrus"
+)
+
+func ParseXRHIdentityHeader(identityHeader string) (*identity.XRHID, error) {
+	var XRHIdentity identity.XRHID
+	decodedIdentity, err := base64.StdEncoding.DecodeString(identityHeader)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding Identity: %v", err)
+	}
+
+	err = json.Unmarshal(decodedIdentity, &XRHIdentity)
+
+	if err != nil {
+		logrus.Errorf("x-rh-identity header is not a valid json: %s. Identity: %s", err.Error(), identityHeader)
+		return nil, fmt.Errorf("x-rh-identity header is not a valid json: %s", err.Error())
+	}
+
+	// XRHIdentity.Identity.User.UserID
+	return &XRHIdentity, nil
+}
+
+type DecodedToken struct {
+	UserId        string `json:"user_id"`
+	OrgId         string `json:"org_id"`
+	AccountNumber string `json:"account_number"`
+	Username      string `json:"username"`
+}
+
+// Parse JWT token without decode key
+func ParseJWTToken(tokenString string) (DecodedToken, error) {
+	str := tokenString
+	str = strings.Split(str, ".")[1]
```

Comment:
question: are you 1000% certain every token passed to this function will contain a period or should we validate len and return an error vs panic?

Reply:
A valid JWT will have follow <header>.<payload>.<signature> format

I copied most of the code from chrome-service-backend: https://github.com/RedHatInsights/chrome-service-backend/blob/main/rest/util/xrh_header_parser.go

I believe it tried and tested for console apps. This middleware is only used when flightctl get deployed on console dot environment

I will add another ParseJWT, which decodes the token using a key and verifies the signature.

Reply:
This is something coming from the external world. We should check the length of the array first to avoid a panic. Return error in case of wrong length.

Reply:
I would consider this blocking merge as it could take down the entire service.